### ISSUE TITLE
document interaction with IntersectionObserver

### ISSUE THREAD

**Issue by fergald** [WICG/display-locking#61]
Based on playing with

https://jsfiddle.net/bpza7vy5/1/

in chromium, it seems like an when an observed element's ancestor locked or unlocked, this triggers an event. I think this is good from the perspective of the virtual scroller, although it would be nice to know from the event that this was the cause, because ideally the virtual scroller would know that it is now inside a locked subtree and there is no point in doing any work at all until that changes.

**Comment by vmpstr** [comment]
We should probably also do resize observer here, which may have similar interactions

**Comment by chrishtr** [comment]
@vmpstr @fergald is the idea to skip IntersectionObserver and ResizeObserver observations for locked subtrees?

**Comment by vmpstr** [comment]
Yeah I think that is the idea. Another point is that (in the current prototype) when switching from visible to invisible and the element is on screen, we issue an intersectionobserver event saying "not intersecting", which I think is correct.

I don't think resize observer needs to do any of this, and just skipping the subtree updates should be enough.

**Comment by chrishtr** [comment]
https://github.com/whatwg/html/issues/4861

**Comment by fergald** [comment]
There are 2 questions (and my suggested answers)

*For intersection*
- what is the behaviour of an observed, locked element?
 - it behaves exactly as an empty div would
- what is the behaviour of an observed element inside a locked subtree
 - it is considered to not intersect any roots and so should never fire any events 
 - when its ancestor transitions into the locked state, its intersection ratio becomes 0% with all roots, that might trigger an event
 - when its ancestor transitions out of the locked state, its intersection ratio can be measured as non-zero, that might trigger an event

I think the above is what we already have.

One weird thing is that when my ancestor locks, the event contains a client rect which has questionable meaning/validity. It could intersect 100% with the observers root. It would be nice to signal that the non-intersection was due to locking. I've opened #69 for this.

 Currently we would just fire an event, including a bounding client rect that presumably reflects where the element _was_. The rect would intersect 100% with the obervser's root which is also a bit odd.
*For resize*

It's almost the same but I would say that an ancestor locking should never cause it to fire but the size at time of locking should be recorded and ancestor unlocking check if that size changed.

I haven't tested the above.
